src/Dialect/ONNX/ShapeInference/Gather.cpp

```diff
@@ -39,14 +39,16 @@ LogicalResult ONNXGatherOpShapeHelper::computeShape(
   // Output shape is constructed from 'input' by:
   //    replacing the dimension at 'axis' in 'input' by the shape of
   //    'indices'.
+  DimsExpr outputDims;
   for (int i = 0; i < dataRank; ++i) {
     if (i == axisIndex)
       for (IndexExpr d : indicesDims)
-        dimsForOutput().emplace_back(d);
+        outputDims.emplace_back(d);
     else
-      dimsForOutput().emplace_back(dataDims[i]);
+      outputDims.emplace_back(dataDims[i]);
   }
 
+  dimsForOutput() = outputDims;
```

Comment:
I don't like that... we should disallow it as now `dimsForOutput` calculate stuff, only to be overridden immediately by the `outputDims`.

Please add a setter to only assign the value.

Reply:
Added a setter `setOutputDims()` to use instead.